dsa/chapter2/bst.py

```diff
@@ -0,0 +1,141 @@
+class BinaryTree:
+    class Node:
```

Comment:
I'd put the `Node` class outside of the `BinaryTree` class. Students might get lost with indentation while reading the code and mix up what code belongs to which class. It also looks better visually in my opinion.